frontend/src/containers/Table.js

```diff
@@ -0,0 +1,268 @@
+        leafs = mapIncluded(item);
+      }
+    });
+
+    leafs.map((item) => {
+      leafsIds = leafsIds.concat(item.id);
+    });
+
+    this.handleSelect(leafsIds);
+  };
+
+  handleSelect = (ids, cb) => {
+    const {
+      updateTableSelection,
+      windowId,
+      viewId,
+      docId,
+      tabId,
+      keyProperty,
+    } = this.props;
+    let newSelected = [];
+
+    if (ids) {
+      if (!ids.splice) {
+        newSelected = [ids];
+      } else {
+        newSelected = ids;
+      }
+    }
+
+    updateTableSelection(
+      getTableId({ windowId, viewId, docId, tabId }),
+      newSelected,
+      keyProperty
+    ).then(() => {
+      cb && cb();
+    });
+
+    return newSelected;
+  };
+
+  handleSelectAll = () => {
+    const { keyProperty, rows } = this.props;
+    const property = keyProperty ? keyProperty : 'rowId';
+    const toSelect = rows.map((item) => item[property]);
+
+    this.handleSelect(toSelect);
+  };
+
+  handleDeselect = (id) => {
+    const { deselectTableItems, windowId, viewId, selected } = this.props;
+    const tableId = getTableId({ windowId, viewId });
+    const index = selected.indexOf(id);
+
+    // TODO: Do we need this returned value ? Maybe we can handle
+    // this in redux only?
+    const newSelected = update(selected, { $splice: [[index, 1]] });
+
+    if (!newSelected.length) {
+      deselectTableItems(tableId, [id]);
+    }
+
+    return newSelected;
+  };
+
+  handleDeselectAll = (callback) => {
+    const { deselectTableItems, windowId, viewId, docId, tabId } = this.props;
+
+    callback && callback();
+
+    deselectTableItems(getTableId({ windowId, viewId, docId, tabId }), []);
+  };
+
+  // TODO: This reallydoesn't do anything. Check if it's still a valid solution
+  handleItemChange = () => {
+    const { onRowEdited } = this.props;
+
+    onRowEdited && onRowEdited(true);
+  };
+
+  openTableModal = () => {
+    const { openModal, windowId, tabId } = this.props;
+
+    openModal('Add new', windowId, 'window', tabId, 'NEW');
+  };
+
+  /**
+   * @method handleAdvancedEdit
+   * @summary Handles advanced edit - i.e case when ALT+E key combinations are being used
+   *          Active only on subtables
+   */
+  handleAdvancedEdit = () => {
+    const { openModal, windowId, tabId, docId, selected } = this.props;
+
+    if (docId) {
+      openModal('Advanced edit', windowId, 'window', tabId, selected[0], true);
+    }
+  };
+
+  handlePromptSubmit = (selected) => {
+    const { windowId, docId, updateDocList, tabId } = this.props;
+
+    this.handleSelect();
+
+    // TODO: This should be an action creator
+    deleteRequest(
+      'window',
+      windowId,
+      docId ? docId : null,
+      docId ? tabId : null,
+      selected
+    ).then(() => {
+      if (!docId) {
+        updateDocList();
+      }
+    });
+  };
+
+  handleZoomInto = (fieldName) => {
+    const { entity, windowId, docId, tabId, viewId, selected } = this.props;
+
+    getZoomIntoWindow(
```

Comment:
I would move this also outside the component.

Reply:
Same as above.